force-app/main/default/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -277,7 +277,14 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
                     numOfContactsLeft = accountIdNumOfCons.get(con.AccountId);
                 }
 
-                updateWrapper.handleUpdateData(con, oldContacts[conProcessed], triggerAction, numOfContactsLeft);
+                Map<Id, Contact> oldContactByContactId = new Map<Id, Contact>(oldContacts);
```

Comment:
This map should be instantiated before the loop, likely at the beginning of the method. Within this code, you are instantiating the map once per contact rather than once at all.

Reply:
That's good catch. On it.